test/translator/verify_test.cc

```diff
@@ -0,0 +1,346 @@
+  };
+  const uint8_t* ptr = reinterpret_cast<const uint8_t*>(&verify_cmd);
+  translator::Span<const uint8_t> scsi_cmd =
+      translator::Span(ptr, sizeof(scsi::Verify10Command));
+  EXPECT_EQ(translator::VerifyToNvme(scsi_cmd, nvme_cmd),
+            translator::StatusCode::kSuccess);
+
+  uint8_t prchk = 0;
+  uint8_t pr_info = 0b1000 | prchk;
+  EXPECT_EQ(nvme_cmd.cdw[0], __bswap_32(verify_cmd.logical_block_address));
+  EXPECT_EQ(nvme_cmd.cdw[1], 0);
+  EXPECT_EQ(nvme_cmd.cdw[2], 536870912);
+}
+
+TEST(Verify, Protect011Bytchk1) {
+  nvme::GenericQueueEntryCmd nvme_cmd;
+  uint32_t lba = 0x12345;
+  const scsi::Verify10Command verify_cmd = {
+      .bytchk = 1,
+      .vr_protect = 0b011,
+      .logical_block_address = htonl(lba),
+      .verification_length = htons(1),
+      .control_byte = {.naca = 0},
+  };
+  const uint8_t* ptr = reinterpret_cast<const uint8_t*>(&verify_cmd);
+  translator::Span<const uint8_t> scsi_cmd =
+      translator::Span(ptr, sizeof(scsi::Verify10Command));
+  EXPECT_EQ(translator::VerifyToNvme(scsi_cmd, nvme_cmd),
+            translator::StatusCode::kSuccess);
+
+  uint8_t prchk = 0;
+  uint8_t pr_info = 0b1000 | prchk;
+  EXPECT_EQ(nvme_cmd.cdw[0], __bswap_32(verify_cmd.logical_block_address));
+  EXPECT_EQ(nvme_cmd.cdw[1], 0);
+  EXPECT_EQ(nvme_cmd.cdw[2], 536870912);
+}
+
+TEST(Verify, Protect100Bytchk1) {
+  nvme::GenericQueueEntryCmd nvme_cmd;
+  uint32_t lba = 0x12345;
+  const scsi::Verify10Command verify_cmd = {
+      .bytchk = 1,
+      .vr_protect = 0b100,
+      .logical_block_address = htonl(lba),
+      .verification_length = htons(1),
+      .control_byte = {.naca = 0},
+  };
+  const uint8_t* ptr = reinterpret_cast<const uint8_t*>(&verify_cmd);
+  translator::Span<const uint8_t> scsi_cmd =
+      translator::Span(ptr, sizeof(scsi::Verify10Command));
+  EXPECT_EQ(translator::VerifyToNvme(scsi_cmd, nvme_cmd),
+            translator::StatusCode::kSuccess);
+
+  uint8_t prchk = 0;
+  uint8_t pr_info = 0b1000 | prchk;
+  EXPECT_EQ(nvme_cmd.cdw[0], __bswap_32(verify_cmd.logical_block_address));
+  EXPECT_EQ(nvme_cmd.cdw[1], 0);
+  EXPECT_EQ(nvme_cmd.cdw[2], 536870912);
+}
+
+TEST(Verify, Protect101Bytchk1) {
+  nvme::GenericQueueEntryCmd nvme_cmd;
+  uint32_t lba = 0x12345;
+  const scsi::Verify10Command verify_cmd = {
+      .bytchk = 1,
+      .vr_protect = 0b101,
+      .logical_block_address = htonl(lba),
+      .verification_length = htons(1),
+      .control_byte = {.naca = 0},
+  };
+  const uint8_t* ptr = reinterpret_cast<const uint8_t*>(&verify_cmd);
+  translator::Span<const uint8_t> scsi_cmd =
+      translator::Span(ptr, sizeof(scsi::Verify10Command));
+  EXPECT_EQ(translator::VerifyToNvme(scsi_cmd, nvme_cmd),
+            translator::StatusCode::kSuccess);
+
+  uint8_t prchk = 0;
+  uint8_t pr_info = 0b1000 | prchk;
+  EXPECT_EQ(nvme_cmd.cdw[0], __bswap_32(verify_cmd.logical_block_address));
+  EXPECT_EQ(nvme_cmd.cdw[1], 0);
+  EXPECT_EQ(nvme_cmd.cdw[2], 536870912);
```

Comment:
When verifying cdw[2], is the result dependent on https://github.com/googleinterns/scsi2nvme/pull/132/files#diff-6b38ddd9bf3769f18b1dfdebbc0425fbR117? If so this logic is not compatible between LE and BE machines. Also, would be nice if we could find a way to make this number slightly less magical

Reply:
I experienced the same issue in my Read PR. The isue with making it less magical is that you end up duplicating the logic that was used to make that field e.g. ` htoll((ntohs(verify_cmd.verification_length) - 1) | ((pr_info) << 26))`. And so that feels like it's not really testing anythng

Reply:
Yeah LE/BE machines will see this number differently. Also, it would be much nicer to store this number as hex. The hex representation is 0x20000000. Much nicer to look at